packages/web-components/fast-element/src/components/element-controller.ts

```diff
@@ -180,22 +189,19 @@ export class ElementController<TElement extends HTMLElement = HTMLElement>
             }
         }
 
-        // Capture any observable values that were set by the binding engine before
-        // the browser upgraded the element. Then delete the property since it will
-        // shadow the getter/setter that is required to make the observable operate.
-        // Later, in the connect callback, we'll re-apply the values.
+        // Delete and re-assign any observable properties that were assigned
+        // before the element is upgraded. This needs to happen because otherwise
+        // the value will shadow the observable getter and setter
         const accessors = Observable.getAccessors(element);
 
         if (accessors.length > 0) {
-            const boundObservables = (this.boundObservables = Object.create(null));
-
             for (let i = 0, ii = accessors.length; i < ii; ++i) {
-                const propertyName = accessors[i].name;
+                const propertyName = accessors[i].name as keyof TElement;
                 const value = (element as any)[propertyName];
 
                 if (value !== void 0) {
-                    delete (element as any)[propertyName];
-                    boundObservables[propertyName] = value;
+                    delete element[propertyName];
+                    element[propertyName] = value;
```

Comment:
It occurs to me that this will cause all the change handlers to fire during the constructor.

Reply:
oops I missed this while reverting, this should be unchanged.

Reply:
huh... I feel like a commit got lost somehow. I'll fix